---
title: C26111
ms.date: 11/04/2016
ms.prod: visual-studio-dev15
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 300f5b4cb5a4534646199ac386e4711cf7e4ebe1
ms.sourcegitcommit: e13e61ddea6032a8282abe16131d9e136a927984
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 04/26/2018
ms.locfileid: "31885114"
---
# <a name="c26111"></a>C26111
Aviso C26111: Falha ao liberar o bloqueio de chamador \<bloqueio > antes de chamar a função \<func >.

 A anotação `_Requires_lock_not_held_` impõe uma pré-condição que a contagem de bloqueio para o bloqueio especificado não pode ser maior que zero quando a função é chamada. C26111 de aviso será emitido quando uma função falha ao liberar o bloqueio antes de chamar outra função.

## <a name="example"></a>Exemplo
 O exemplo a seguir gera o aviso C26111 porque o `_Requires_lock_not_held_` pré-condição for violada pela chamada para `DoNotLock` dentro da seção bloqueada.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}

```